vhd/vhd-index.c

```diff
@@ -513,8 +513,11 @@ vhd_index_clone_bat(vhdi_name_t *name, const char *parent)
 	memset(&files, 0, sizeof(vhdi_file_table_t));
 
 	err = asprintf(&pbat, "%s.bat", parent);
-	if (err == -1)
+	if (err == -1){
+		free(pbat);
+		pbat = NULL;
```

Comment:
This particular line is not helpful.  pbat is a local scope variable, and there is no chance of it being available after the return statement